### PR TITLE
feature/scoop-controller

### PR DESCRIPTION
**Pull request by nathantfrank**
Scoop Controller Implementation

### CODE REVIEW

File: cr17/nodes/cr17/scoop_controller.py [hunk 1]
```diff
@@ -1,15 +1,103 @@
 #!/usr/bin/env python2
 import rospy
+from cr17.msg import scoopControl
+from cr17.srv import autonomousActive
+
+# Topics
+ARM_STATE_TOPIC = rospy.get_param('topics/scoop_state_cmds')
```
Comment (bgrayd) [inline cr17/nodes/cr17/scoop_controller.py]: This should be added to the parameter file topics.yaml

File: cr17/nodes/cr17/scoop_controller.py [hunk 2]
```diff
@@ -1,15 +1,103 @@
...
+ARM_GOAL_TOPIC = rospy.get_param('topics/scoop_cmds')
```
Comment (bgrayd) [inline cr17/nodes/cr17/scoop_controller.py]: This should be added to the parameter file topics.yaml

**Review by bgrayd — CHANGES_REQUESTED** [review]

File: cr17/nodes/cr17/scoop_controller.py [hunk 3]
```diff
@@ -1,15 +1,103 @@
...
+                    new_msg.armAngle = self.scoop_msg.armAngle
+                    new_msg.scoopAngle = self.scoop_msg.scoopAngle
+                    self.scoop_pub.publish(new_msg)
+                else:
+                    self.scoop_pub.publish(self.defined_msgs[self.scoop_msg.desiredState])
+            rate.sleep()
+
+    def set_autonomy(self, req):
+        self.autonomous = req.autonomousActive
+        req.acknowledged = True
```
Comment (bgrayd) [inline cr17/nodes/cr17/scoop_controller.py]: This causes an error.  The req variable is just the request part of the service call.  For setting the output of this service, autonomousActiveResponse should be instantiated and used.

File: cr17/nodes/cr17/scoop_controller.py [hunk 4]
```diff
@@ -1,15 +1,103 @@
 #!/usr/bin/env python2
 import rospy
+from cr17.msg import scoopControl
+from cr17.srv import autonomousActive
```
Comment (bgrayd) [inline cr17/nodes/cr17/scoop_controller.py]: Should also import autonomousActiveResponse from the services as well

**Review by bgrayd — DISMISSED** [review]

**Comment by JacobAMason** [comment]
@bgrayd Can you give this an updated review so we can merge it?

File: cr17/nodes/cr17/high_level_state_controller.py [hunk 5]
```diff
@@ -5,8 +5,6 @@
 from geometry_msgs.msg import PoseStamped, Twist, Pose
 from std_msgs.msg import Float64, Bool, String
 from nav_msgs.msg import Path
-from cr17.msg import scoopControl
```
Comment (bgrayd) [inline cr17/nodes/cr17/high_level_state_controller.py]: Why is this being removed?

File: cr17/nodes/cr17/high_level_state_controller.py [hunk 6]
```diff
@@ -5,8 +5,6 @@
...
-from cr17.srv import autonomousActive, autonomousActiveResponse
```
Comment (bgrayd) [inline cr17/nodes/cr17/high_level_state_controller.py]: I think something might be going wrong with the merge on this file...

File: cr17/nodes/cr17/high_level_state_controller.py [hunk 7]
```diff
@@ -40,32 +34,65 @@ def __init__(self):
...
         self.autostate = "INIT"
+        self.highState = "AUTO"
+        self.lastJoy = None
+        self.leftDrive = None
+        self.rightDrive = None
+        self.armDown = None
+        self.armUp = None
+        self.handDown = None
+        self.handUp = None
+        self.teleopButton_prev = 0
```
Comment (bgrayd) [inline cr17/nodes/cr17/high_level_state_controller.py]: Yea, this is not the correct high_level_state_controller, this merge will wipe lots of changes to it

**Review by bgrayd — CHANGES_REQUESTED** [review]

**Comment by nathantfrank** [comment]
@bgrayd the branch should be correctly up to date with develop now.

**Review by bgrayd — APPROVED** [review]
LGTM

**Pull request merged**